momento-sdk/src/main/java/momento/sdk/SubscriptionWrapper.java

```diff
@@ -18,21 +17,23 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class SubscriptionWrapper implements Closeable {
+class SubscriptionWrapper implements AutoCloseable {
   private final Logger logger = LoggerFactory.getLogger(SubscriptionWrapper.class);
-  private final ScsTopicGrpcStubsManager grpcManager;
+  private final IScsTopicConnection connection;
   private final SendSubscribeOptions options;
   private boolean firstMessage = true;
+  private boolean isConnectionLost = false;
+
   private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
 
   private CancelableClientCallStreamObserver<_SubscriptionItem> subscription;
 
-  SubscriptionWrapper(ScsTopicGrpcStubsManager grpcManager, SendSubscribeOptions options) {
-    this.grpcManager = grpcManager;
+  SubscriptionWrapper(IScsTopicConnection connection, SendSubscribeOptions options) {
+    this.connection = connection;
     this.options = options;
   }
 
-  CompletableFuture<Void> subscribeWithRetry() {
+  public CompletableFuture<Void> subscribeWithRetry() {
```

Comment:
can you add a docstring that indicates this is only visible for the purposes of testing, not intended to be called except by the topic client?